.devcontainer/Containerfile

```diff
@@ -6,14 +6,26 @@ RUN echo -e "\\n\\n# Added during image build" >> /etc/dnf/dnf.conf && \
     echo -e "minrate=100\\ntimeout=60\\n" >> /etc/dnf/dnf.conf
 
 # Install dependencies for the dev environment
-ARG INSTALL_RPMS="podman openssh-clients cpp git-core golang which procps-ng curl helm kubernetes1.33-client jq hadolint gh"
+ARG INSTALL_RPMS="podman openssh-clients cpp git-core gcc which procps-ng curl helm kubernetes1.33-client jq hadolint gh"
 RUN dnf -y makecache && \
     dnf -y update && \
     dnf -y install $INSTALL_RPMS --exclude container-selinux && \
     dnf -y install glibc-debuginfo gcc-debuginfo libgcc-debuginfo --enablerepo=fedora-debuginfo,updates-debuginfo && \
     dnf clean all && \
     rm -rf /var/cache /var/log/dnf* /var/log/yum.*
 
+# Install Go (version-locked)
+ARG GO_VERSION=1.25.3
```

Comment:
We're getting quite a lot of issues in renovate's PRs for go dependencies as the go version is being set in multiple contexts (e.g. go.mod, workflows files ). Here we add another one. Not sure what's the way to handle it, but it's not that trivial to maintain.

What was the reason we